examples/gpu_functions/reduction.mojo

```diff
@@ -0,0 +1,109 @@
+# ===----------------------------------------------------------------------=== #
+# Copyright (c) 2025, Modular Inc. All rights reserved.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions:
+# https://llvm.org/LICENSE.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===----------------------------------------------------------------------=== #
+
+from math import ceildiv
+from memory import UnsafePointer
+from os.atomic import Atomic
+from random import randint
+from sys import has_amd_gpu_accelerator, has_nvidia_gpu_accelerator
+from testing import assert_equal
+
+from gpu import thread_idx, block_idx, block_dim, grid_dim, warp, barrier
+from gpu.host import DeviceContext
+from gpu.memory import load
+from layout.tensor_builder import LayoutTensorBuild as tb
+
+# Initialize parameters
+# To archieve high bandwidth increase SIZE to large value
+alias TPB = 512
+alias LOG_TPB = 9
+alias BATCH_SIZE = 8  # needs to be power of 2
+alias SIZE = 1 << 25
+alias NUM_BLOCKS = ceildiv(SIZE, TPB * BATCH_SIZE)
+alias dtype = DType.int32
+
+fn sum_kernel[
+    size: Int, batch_size: Int
+](out: UnsafePointer[Int32], a: UnsafePointer[Int32],):
```

Comment:
Since you're already using `LayoutTensorBuild` then would be great to use the LayoutTensor over the UnsafePointer for the better consistency.